--- overlays/wlogout/main.c.orig	2026-04-25 16:25:34.400107844 +1200
+++ overlays/wlogout/main.c	2026-04-25 01:06:16.274741699 +1200
@@ -623,6 +623,24 @@
                                          NULL);
 }
 
+static gboolean on_button_enter(GtkWidget *widget, GdkEventCrossing *event, gpointer data)
+{
+    gtk_widget_grab_focus(widget);
+
+    GdkDisplay *display = gtk_widget_get_display(widget);
+    GdkCursor *cursor = gdk_cursor_new_from_name(display, "pointer");
+    gdk_window_set_cursor(gtk_widget_get_window(widget), cursor);
+    g_object_unref(cursor);
+
+    return FALSE;
+}
+
+static gboolean on_button_leave(GtkWidget *widget, GdkEventCrossing *event, gpointer data)
+{
+    gdk_window_set_cursor(gtk_widget_get_window(widget), NULL);
+    return FALSE;
+}
+
 static void load_buttons(GtkContainer *container)
 {
     GtkWidget *grid = gtk_grid_new();
@@ -660,6 +678,8 @@
                 strcat(buttons[count].text, "]");
             }
             but[i][j] = gtk_button_new_with_label(buttons[count].text);
+            g_signal_connect(but[i][j], "enter-notify-event", G_CALLBACK(on_button_enter), NULL);
+            g_signal_connect(but[i][j], "leave-notify-event", G_CALLBACK(on_button_leave), NULL);
             gtk_widget_set_name(but[i][j], buttons[count].label);
             gtk_label_set_yalign(
                 GTK_LABEL(gtk_bin_get_child(GTK_BIN(but[i][j]))),
